crates/exex/src/hints.rs

```diff
@@ -128,3 +156,119 @@ pub fn print_tx_hint() -> Hint {
         },
     )
 }
+
+/// Generates a hint to store block information in the `Environment` model.
+pub fn block_info_hint(block: SealedBlock, transaction: TransactionSignedEcRecovered) -> Hint {
+    Hint::new(
+        String::from("block_info"),
+        move |vm: &mut VirtualMachine,
+              _exec_scopes: &mut ExecutionScopes,
+              ids_data: &HashMap<String, HintReference>,
+              ap_tracking: &ApTracking,
+              _constants: &HashMap<String, Felt252>|
+              -> Result<(), HintError> {
+            // We retrieve the `env` pointer from the `ids_data` hashmap.
+            // This pointer is used to store the block-related values in the VM.
+            let env_ptr = get_ptr_from_var_name("env", vm, ids_data, ap_tracking)?;
+
+            // Define memory addresses for each block-related value by incrementing the `env_ptr`.
+            // The first first field of the Cairo struct is stored at the lowest memory address.
+            // Then, the next field is stored at the next memory address, and so on.
+            let origin_addr = (env_ptr + 0_i32)?;
+            let gas_price_addr = (env_ptr + 1_i32)?;
+            let chain_id_addr = (env_ptr + 2_i32)?;
+            let prev_randao_low_addr = (env_ptr + 3_i32)?;
+            let prev_randao_high_addr = (env_ptr + 4_i32)?;
+            let block_number_addr = (env_ptr + 5_i32)?;
+            let block_gas_limit_addr = (env_ptr + 6_i32)?;
+            let block_timestamp_addr = (env_ptr + 7_i32)?;
+            let coinbase_addr = (env_ptr + 8_i32)?;
+            let base_fee_addr = (env_ptr + 9_i32)?;
+
+            // Insert the transaction origin (signer) into the VM memory.
+            let _ = vm
+                .insert_value(origin_addr, Felt252::from_bytes_be_slice(&transaction.signer().0 .0))
+                .map_err(HintError::Memory);
+
+            // Insert the gas price of the transaction into the VM memory.
+            let _ = vm
+                .insert_value(
+                    gas_price_addr,
+                    Felt252::from(transaction.effective_gas_price(block.base_fee_per_gas)),
+                )
+                .map_err(HintError::Memory);
+
+            // Insert the chain ID of the transaction, defaulting to 0 if not available.
+            let _ = vm
```

Comment:
Isn’t there any write_segments like in python to alleviate a bit the function here ?

Reply:
Mhhh don't think so. I had a look at similar examples in the rust cairo-vm and I think they use a similar approach:

https://github.com/lambdaclass/cairo-vm/blob/1f1190b5c079c777c41e35c348eaa0b5447063e8/vm/src/hint_processor/builtin_hint_processor/secp/bigint_utils.rs#L80

https://github.com/lambdaclass/cairo-vm/blob/1f1190b5c079c777c41e35c348eaa0b5447063e8/vm/src/hint_processor/builtin_hint_processor/keccak_utils.rs#L110

And inside the vm core I think this is the only function available to insert values in memory segments, wdyt?

https://github.com/lambdaclass/cairo-vm/blob/1f1190b5c079c777c41e35c348eaa0b5447063e8/vm/src/vm/vm_core.rs

Reply:
I think that https://github.com/lambdaclass/cairo-vm/blob/main/vm/src/vm/vm_memory/memory_segments.rs#L68 should do it

Reply:
Oh yes true didn't see this one, just modified